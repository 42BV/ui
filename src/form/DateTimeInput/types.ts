// See https://momentjs.com/docs/#/displaying/format/ for formatting options.
export type DateFormat = string | false; // When false it should not be used
export type TimeFormat = string | false; // When false it should not be used
