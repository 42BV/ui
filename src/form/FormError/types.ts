export type OnChange = (hasErrors: boolean) => void;