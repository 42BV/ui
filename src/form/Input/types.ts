export type Position = 'left' | 'right';
export type Mask = (string | RegExp)[];
