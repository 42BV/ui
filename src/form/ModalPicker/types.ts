export type AddButtonCallback<T> = () => Promise<T>;

export interface AddButtonOptions<T> {
  label: string;
  onClick: AddButtonCallback<T>;
}
