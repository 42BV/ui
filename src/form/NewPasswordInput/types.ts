export type NewPasswordInputRule =
  | 'lowercase'
  | 'uppercase'
  | 'number'
  | 'specialChar'
  | 'minimumLength'
  | 'noSpace';
