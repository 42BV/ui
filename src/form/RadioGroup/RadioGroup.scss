.radio-group > legend {
  font-size: 1rem;
}
