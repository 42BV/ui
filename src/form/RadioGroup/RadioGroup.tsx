import React from 'react';
import { FormGroup, Input, Label } from 'reactstrap';
import { constant, get } from 'lodash';

import withJarb from '../withJarb/withJarb';
import { Color } from '../types';
import { t } from '../../utilities/translation/translation';
import {
  isOptionSelected,
  OptionEnabledCallback,
  OptionEqual,
  OptionForValue,
  OptionsFetcher
} from '../option';
import { doBlur } from '../utils';
import Loading from '../../core/Loading/Loading';
import { useOptions } from '../useOptions';

export interface Text {
  /**
   * The message to show when the select is loading. Defaults
   * to `loading...`
   */
  loadingMessage?: string;
}

interface BaseProps<T> {
  /**
   * The id of the form element.
   */
  id?: string;

  /**
   * The value that the form element currently has.
   */
  value?: T;

  /**
   * Callback for when the form element changes.
   */
  onChange: (value: T) => void;

  /**
   * Optional callback for when the form element is blurred.
   */
  onBlur?: () => void;

  /**
   * Is either an array of options or a callback which fetches
   * the options asynchronously.
   */
  options: OptionsFetcher<T> | T[];

  /**
   * Callback to convert an value of type T to an option to show
   * to the user.
   */
  optionForValue: OptionForValue<T>;

  /**
   * Optional callback which is used to determine if two options
   * of type T are equal.
   *
   * When `isOptionEqual` is not defined the outcome of `optionForValue`
   * is used to test equality.
   */
  isOptionEqual?: OptionEqual<T>;

  /**
   * Optional callback which is called for every option to determine
   * if the option can be selected. By default all options can be
   * selected.
   */
  isOptionEnabled?: OptionEnabledCallback<T>;

  /**
   * Optionally the error message to render.
   */
  error?: React.ReactNode;

  /**
   * Optionally the color of the FormGroup.
   */
  color?: Color;

  /**
   * Whether or not the form element is currently valid.
   */
  valid?: boolean;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;

  /**
   * The placeholder of the form element.
   */
  placeholder?: string;

  /**
   * Optionally customized text within the component.
   * This text should already be translated.
   */
  text?: Text;
}

interface WithoutLabel<T> extends BaseProps<T> {
  label?: never;
}

interface WithLabel<T> extends BaseProps<T> {
  /**
   * The label of the form element.
   */
  label: string;
}

export type Props<T> = WithoutLabel<T> | WithLabel<T>;

/**
 * RadioGroup is a form element for which the value can be selected
 * from a limited range.
 */
export default function RadioGroup<T>(props: Props<T>) {
  const {
    value,
    error,
    color,
    text = {},
    className = '',
    placeholder,
    onChange,
    onBlur,
    optionForValue,
    isOptionEqual
  } = props;

  const { options, loading } = useOptions({
    optionsOrFetcher: props.options,
    value,
    isOptionEqual,
    optionForValue
  });

  const isOptionEnabled = get(props, 'isOptionEnabled', constant(true));

  function onRadioClicked(option: T) {
    onChange(option);
    doBlur(onBlur);
  }

  return (
    <FormGroup tag="fieldset" className={className} color={color}>
      {'label' in props && props.label ? <legend>{props.label}</legend> : null}
      {placeholder ? (
        <p className="text-muted">
          <em>{placeholder}</em>
        </p>
      ) : null}
      {loading ? (
        <Loading>
          {t({
            key: 'RadioGroup.LOADING',
            fallback: 'Loading...',
            overrideText: text.loadingMessage
          })}
        </Loading>
      ) : (
        options.map(option => {
          const label = optionForValue(option);

          return (
            <FormGroup key={label} check>
              <Label check>
                <Input
                  type="radio"
                  value={label}
                  checked={isOptionSelected({
                    option,
                    optionForValue,
                    isOptionEqual,
                    value
                  })}
                  disabled={!isOptionEnabled(option)}
                  onChange={() => onRadioClicked(option)}
                />{' '}
                {label}
              </Label>
            </FormGroup>
          );
        })
      )}

      {error}
    </FormGroup>
  );
}
/**
 * Variant of the RadioGroup which can be used in a Jarb context.
 */
export const JarbRadioGroup = withJarb<any, any, Props<any>>(RadioGroup);
