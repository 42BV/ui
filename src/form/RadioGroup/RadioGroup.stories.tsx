import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';

import { JarbRadioGroup, RadioGroup } from './RadioGroup';
import {
  FinalForm,
  IsOptionEqualInfo,
  JarbFormElementDependencies,
  KeyForOptionInfo,
  nonExistingProvince,
  Province,
  provinceFetcher,
  provinces,
  ReloadOptionsInfo,
  resolveAfter
} from '../story-utils';
import { Card } from '../../core/Card/Card';
import { pageOf } from '../../utilities/page/page';
import { Tooltip } from '../../core/Tooltip/Tooltip';
import { Icon } from '../../core/Icon';

storiesOf('Form/RadioGroup', module)
  .addParameters({ component: RadioGroup })
  .add('predefined options', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })
  .add('async options', () => {
    const [ value, setValue ] = useState<Province | undefined>(provinces()[0]);

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinceFetcher}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <p className="mt-3">
          <strong>Disclaimer:</strong> when using async, a maximum of 100
          options will be displayed. If you want to display more than 100
          options, you should use the ModalPickerSingle.
        </p>
      </Card>
    );
  })
  .add('disabled options', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          isOptionEnabled={(option) => !option.label.startsWith('Z')}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })
  .add('custom isOptionEqual', () => {
    const [ value, setValue ] = useState<Province | undefined>(provinces()[0]);

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          isOptionEqual={(a, b) => a.value === b.value}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <IsOptionEqualInfo />
      </Card>
    );
  })
  .add('custom keyForOption', () => {
    const [ value, setValue ] = useState<Province | undefined>(provinces()[0]);

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          keyForOption={(province) => province.value}
          value={value}
          onChange={setValue}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <KeyForOptionInfo />
      </Card>
    );
  })
  .add('using reloadOptions', () => {
    const [ brand, setBrand ] = useState<string>();
    const [ model, setModel ] = useState<string>();

    const allOptions = {
      Audi: [ 'A1', 'A2', 'A3', 'M5' ],
      BMW: [ 'series 1', 'series 2', 'series 3', 'series 4', 'series 5' ],
      Mercedes: [ 'Viano', 'Vito', 'Sprinter' ]
    };

    return (
      <Card className="m-2">
        <RadioGroup
          id="brand"
          label="Brand"
          placeholder="Please select your brand"
          options={() => resolveAfter(pageOf(Object.keys(allOptions), 1))}
          labelForOption={(option) => option}
          value={brand}
          onChange={(value) => {
            setBrand(value);
            setModel(undefined);
          }}
        />
        <RadioGroup
          id="model"
          label="Model"
          placeholder={
            brand ? 'Please select your model' : 'Please select a brand first'
          }
          options={() =>
            resolveAfter(pageOf(brand ? allOptions[brand] : [], 1))
          }
          labelForOption={(option: string) => option}
          value={model}
          onChange={setModel}
          reloadOptions={brand}
        />

        {brand ? <p>Your chosen brand is: {brand}</p> : null}
        {model ? <p>Your chosen model is: {model}</p> : null}

        <ReloadOptionsInfo />
      </Card>
    );
  })
  .add('label & placeholder', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <span className="d-block fs-5">Invisible label</span>

        <RadioGroup<Province>
          label="Province"
          hiddenLabel={true}
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        <hr />

        <span className="d-block fs-5">Custom label</span>

        <RadioGroup<Province>
          id="provinces"
          label={
            <div className="d-flex justify-content-between">
              <span>Subject</span>
              <Tooltip className="ms-1" content="The province you live in">
                <Icon icon="info" />
              </Tooltip>
            </div>
          }
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        <hr />

        <span className="d-block fs-5">Without placeholder</span>

        <RadioGroup<Province>
          id="province"
          label="Province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
        />

        <hr />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })
  .add('horizontal', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
          horizontal={true}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <p>
          <strong>Disclaimer:</strong> horizontal mode works best when there are
          not too many items
        </p>
      </Card>
    );
  })
  .add('horizontal with clear', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
          horizontal={true}
          canClear={true}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}

        <p>
          <strong>Disclaimer:</strong> horizontal mode works best when there are
          not too many items
        </p>
      </Card>
    );
  })
  .add('with clear button', () => {
    const [ value, setValue ] = useState<Province | undefined>(
      nonExistingProvince()
    );

    return (
      <Card className="m-2">
        <RadioGroup<Province>
          id="province"
          label="Province"
          placeholder="Please select your province"
          options={provinces()}
          labelForOption={(province) => province.label}
          value={value}
          onChange={setValue}
          canClear={true}
        />

        {value ? <p>Your chosen province is: {value.label}</p> : null}
      </Card>
    );
  })

  .add('jarb', () => {
    return (
      <>
        <JarbFormElementDependencies />
        <FinalForm>
          <JarbRadioGroup
            id="province"
            name="province"
            label="Province"
            placeholder="Please select your province"
            options={provinceFetcher}
            labelForOption={(province) => province.label}
            jarb={{
              validator: 'User.province',
              label: 'Province'
            }}
          />
        </FinalForm>
      </>
    );
  });
