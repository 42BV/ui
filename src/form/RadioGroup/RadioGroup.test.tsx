import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import RadioGroup, { Text } from './RadioGroup';
import { User } from '../../test/types';
import { adminUser, coordinatorUser, userUser } from '../../test/fixtures';
import { OptionEnabledCallback } from '../option';

describe('Component: RadioGroup', () => {
  let radioGroup: ShallowWrapper;

  let onChangeSpy: jest.Mock<any, any>;
  let onBlurSpy: jest.Mock<any, any>;

  function setup({
    value,
    isOptionEnabled,
    text,
    hasPlaceholder = true
  }: {
    value?: User;
    isOptionEnabled?: OptionEnabledCallback<User>;
    text?: Text;
    hasPlaceholder?: boolean;
  }) {
    onChangeSpy = jest.fn();
    onBlurSpy = jest.fn();

    radioGroup = shallow(
      <RadioGroup
        id="subject"
        label="Subject"
        placeholder={hasPlaceholder ? 'Please enter your subject' : undefined}
        text={text}
        isOptionEnabled={isOptionEnabled}
        optionForValue={(user: User) => user?.email}
        options={[adminUser, coordinatorUser, userUser]}
        value={value}
        onChange={onChangeSpy}
        onBlur={onBlurSpy}
        error="Some error"
        valid={false}
      />
    );
  }

  describe('ui', () => {
    test('with value', () => {
      setup({ value: adminUser, isOptionEnabled: undefined });

      expect(toJson(radioGroup)).toMatchSnapshot(
        'Component: RadioGroup => ui => with value'
      );
    });

    test('when value is not in options select nothing', () => {
      setup({
        value: {
          id: -1,
          email: 'none',
          active: false,
          roles: []
        },
        isOptionEnabled: undefined
      });

      const radios = radioGroup.find('Input');

      expect(radios.length).toBe(3);

      expect(radios.at(0).props().checked).toBe(false);
      expect(radios.at(1).props().checked).toBe(false);
      expect(radios.at(2).props().checked).toBe(false);
    });

    test('without placeholder', () => {
      setup({
        value: adminUser,
        isOptionEnabled: undefined,
        hasPlaceholder: false
      });

      expect(toJson(radioGroup)).toMatchSnapshot(
        'Component: RadioGroup => ui => without placeholder'
      );
    });
  });

  describe('events', () => {
    test('onChange', () => {
      setup({ value: undefined, isOptionEnabled: undefined });

      const radio = radioGroup.find('Input').at(0);

      // @ts-ignore
      radio.props().onChange({ target: { value: 0 } });

      expect(onChangeSpy).toHaveBeenCalledTimes(1);
      expect(onChangeSpy).toHaveBeenCalledWith(adminUser);
    });

    describe('isOptionEnabled', () => {
      it('should when "isOptionEnabled" is not defined always be true', () => {
        setup({ value: adminUser, isOptionEnabled: undefined });

        const radios = radioGroup.find('Input');
        expect(radios.length).toBe(3);

        // Other options should be enabled
        expect(radios.at(0).props().disabled).toBe(false);
        expect(radios.at(1).props().disabled).toBe(false);
        expect(radios.at(2).props().disabled).toBe(false);
      });

      it('should when "isOptionEnabled" is defined use that to determine if the option is enabled', () => {
        const isOptionEnabledSpy = jest.fn();

        // Disabled all option now
        isOptionEnabledSpy.mockReturnValue(false);

        setup({ value: undefined, isOptionEnabled: isOptionEnabledSpy });

        const radios = radioGroup.find('Input');
        expect(radios.length).toBe(3);

        // Other options should be enabled
        expect(radios.at(0).props().disabled).toBe(true);
        expect(radios.at(1).props().disabled).toBe(true);
        expect(radios.at(2).props().disabled).toBe(true);

        expect(isOptionEnabledSpy).toHaveBeenCalledTimes(3);
        expect(isOptionEnabledSpy.mock.calls).toEqual([
          [adminUser],
          [coordinatorUser],
          [userUser]
        ]);
      });
    });
  });

  describe('value changes', () => {
    test('becomes empty', () => {
      setup({ value: userUser, isOptionEnabled: undefined });

      let radios = radioGroup.find('Input');

      expect(radios.at(0).props().checked).toBe(false);
      expect(radios.at(1).props().checked).toBe(false);
      expect(radios.at(2).props().checked).toBe(true);

      radioGroup.setProps({ value: undefined });

      radios = radioGroup.find('Input');

      expect(radios.at(0).props().checked).toBe(false);
      expect(radios.at(1).props().checked).toBe(false);
      expect(radios.at(2).props().checked).toBe(false);
    });

    test('becomes filled', () => {
      setup({ value: undefined, isOptionEnabled: undefined });

      let radios = radioGroup.find('Input');

      expect(radios.at(0).props().checked).toBe(false);
      expect(radios.at(1).props().checked).toBe(false);
      expect(radios.at(2).props().checked).toBe(false);

      radioGroup.setProps({ value: coordinatorUser });

      radios = radioGroup.find('Input');

      expect(radios.at(0).props().checked).toBe(false);
      expect(radios.at(1).props().checked).toBe(true);
      expect(radios.at(2).props().checked).toBe(false);
    });
  });
});
