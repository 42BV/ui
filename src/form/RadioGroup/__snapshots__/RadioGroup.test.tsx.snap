// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RadioGroup ui horizontal: Component: RadioGroup => ui => horizontal 1`] = `
<FormGroup
  className="radio-group "
  tag="fieldset"
>
  <legend>
    Subject
  </legend>
  <p
    className="text-muted"
  >
    <em>
      Please enter your subject
    </em>
  </p>
  <FormGroup
    check={true}
    inline={true}
    key="42"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={true}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="admin@42.nl"
      />
       
      admin@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={true}
    key="777"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="coordinator@42.nl"
      />
       
      coordinator@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={true}
    key="1337"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="user@42.nl"
      />
       
      user@42.nl
    </Label>
  </FormGroup>
  Some error
</FormGroup>
`;

exports[`Component: RadioGroup ui loading: Component: RadioGroup => ui => loading 1`] = `
<FormGroup
  className="radio-group "
  tag="fieldset"
>
  <legend>
    Subject
  </legend>
  <p
    className="text-muted"
  >
    <em>
      Please enter your subject
    </em>
  </p>
  <Loading>
    Loading...
  </Loading>
  Some error
</FormGroup>
`;

exports[`Component: RadioGroup ui with value: Component: RadioGroup => ui => with value 1`] = `
<FormGroup
  className="radio-group "
  tag="fieldset"
>
  <legend>
    Subject
  </legend>
  <p
    className="text-muted"
  >
    <em>
      Please enter your subject
    </em>
  </p>
  <FormGroup
    check={true}
    inline={false}
    key="42"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={true}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="admin@42.nl"
      />
       
      admin@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={false}
    key="777"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="coordinator@42.nl"
      />
       
      coordinator@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={false}
    key="1337"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="user@42.nl"
      />
       
      user@42.nl
    </Label>
  </FormGroup>
  Some error
</FormGroup>
`;

exports[`Component: RadioGroup ui without label: Component: RadioGroup => ui => without label 1`] = `
<FormGroup
  className="radio-group "
  tag="fieldset"
>
  <p
    className="text-muted"
  >
    <em>
      Please enter your subject
    </em>
  </p>
  <FormGroup
    check={true}
    inline={false}
    key="42"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={true}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="admin@42.nl"
      />
       
      admin@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={false}
    key="777"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="coordinator@42.nl"
      />
       
      coordinator@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={false}
    key="1337"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="user@42.nl"
      />
       
      user@42.nl
    </Label>
  </FormGroup>
  Some error
</FormGroup>
`;

exports[`Component: RadioGroup ui without placeholder: Component: RadioGroup => ui => without placeholder 1`] = `
<FormGroup
  className="radio-group "
  tag="fieldset"
>
  <legend>
    Subject
  </legend>
  <FormGroup
    check={true}
    inline={false}
    key="42"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={true}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="admin@42.nl"
      />
       
      admin@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={false}
    key="777"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="coordinator@42.nl"
      />
       
      coordinator@42.nl
    </Label>
  </FormGroup>
  <FormGroup
    check={true}
    inline={false}
    key="1337"
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="user@42.nl"
      />
       
      user@42.nl
    </Label>
  </FormGroup>
  Some error
</FormGroup>
`;
